### PR TITLE
Create janus api client

### PR DESCRIPTION
**Pull request by arothuis**
[Ticket 104427028](https://www.pivotaltracker.com/story/show/104427028)

### CODE REVIEW

**Comment by arothuis** [comment]
@rjkip @DRvanR

File: src/Surfnet/JanusApiClientBundle/Service/ApiService.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        ];
+
+        $data = json_decode($json, true);
+
+        if (JSON_ERROR_NONE !== json_last_error()) {
+            $last = json_last_error();
+
+            throw new CoreInvalidArgumentException(
+                'Unable to parse JSON data: '
+                . (isset($jsonErrors[$last])
```
Comment (DRvanR) [inline src/Surfnet/JanusApiClientBundle/Service/ApiService.php]: personally I'd place the ternary outside of the string construction, makes it a bit more readable.

**Comment by DRvanR** [comment]
Minor nitpick, the rest seems ok, :+1: from me.

File: src/Surfnet/JanusApiClientBundle/Exception/InvalidResponseException.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\JanusApiClientBundle\Exception;
+
+use RuntimeException as CoreRuntimeException;
+
+class InvalidResponseException extends CoreRuntimeException implements SurfnetJanusApiClientException
```
Comment (rjkip) [inline src/Surfnet/JanusApiClientBundle/Exception/InvalidResponseException.php]: Could you insert our own RuntimeException in the class hierarchy (which extends CoreRuntimeException)? Nice use of a "package-level" interface :+1:

File: src/Surfnet/JanusApiClientBundle/Service/ApiService.php [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param $path
+     * @param array $parameters
+     * @return string
+     * @throw CoreRuntimeException
```
Comment (rjkip) [inline src/Surfnet/JanusApiClientBundle/Service/ApiService.php]: `@throws`

**Comment by rjkip** [comment]
Minor nitpick, the rest seems ok, :+1: from me.

dejavu

File: src/Surfnet/JanusApiClientBundle/Exception/InvalidResponseException.php (hunk 2 above)
Comment (arothuis) [inline src/Surfnet/JanusApiClientBundle/Exception/InvalidResponseException.php]: Will do. Thanks, merely following a great example.

**Comment by arothuis** [comment]
@DRvanR @rjkip

**Comment by rjkip** [comment]
:+1:

**Pull request merged**